MewsSystems/developers
# FE Movie Search: add movie search and movie detail views

### Summary
This pull request adds a movie search application as part of the Mews frontend developer task. The application consists of two views: a search view and a movie detail view.

### Features
1. **Search View**
   - A search input that automatically fetches movies from the TMDB API after typing stops.
   - Paginated list of movies with the ability to load additional results.
   - Loading indicator during API requests.
   - Error and "No Results" handling for a better user experience.

2. **Movie Detail View**
   - Displays detailed information about a selected movie, including title, overview, release date, and rating.

### How to Test
1. Clone this branch.
2. Run `yarn start` it should open the search page
3. Type on the input any movie title: e.g. Lord of the Rings
4. Movies should appear when you stop typing
5. Click on any movie and Detal Page should appear 

### API Integration
Movies are fetched from [The Movie Database (TMDB) API](https://www.themoviedb.org/), using the provided API key `03b8572954325680265531140190fd2a`.

### Screenshots

| **View**        | **Screenshot**                       |
|------------------|-------------------------------------|
| **Search View**  | <img width="1544" alt="Captura de pantalla 2024-12-17 a las 21 59 31" src="https://github.com/user-attachments/assets/eacdcebd-0c7e-4888-92d8-95dea6e8fbbe" /> |
| **Detail View**  | <img width="1784" alt="Captura de pantalla 2024-12-17 a las 22 00 08" src="https://github.com/user-attachments/assets/abad6d4f-b98b-43ac-b91d-5c71367d77dc" /> |
